Data/models/TabMO.swift

```diff
@@ -69,50 +69,61 @@ public final class TabMO: NSManagedObject, CRUD {
+            tab.title = Strings.New_Tab
+        }
+        
+        return uuidString
     }
 
     // Updates existing tab with new data. Usually called when user navigates to a new website for in his existing tab.
-    @discardableResult public class func update(tabData: SavedTab) -> TabMO? {
-        let context = DataController.newBackgroundContext()
-        guard let tab = get(fromId: tabData.id, context: context) else { return nil }
-        
-        if let screenshot = tabData.screenshot {
-            tab.screenshot = UIImageJPEGRepresentation(screenshot, 1)
+    public class func update(tabData: SavedTab) {
+        DataController.performTask { context in
+            guard let tabToUpdate = get(fromId: tabData.id, context: context) else { return }
+            
+            if let screenshot = tabData.screenshot {
+                tabToUpdate.screenshot = UIImageJPEGRepresentation(screenshot, 1)
+            }
+            tabToUpdate.url = tabData.url
+            tabToUpdate.order = tabData.order
+            tabToUpdate.title = tabData.title
+            tabToUpdate.urlHistorySnapshot = tabData.history as NSArray
+            tabToUpdate.urlHistoryCurrentIndex = tabData.historyIndex
+            tabToUpdate.isSelected = tabData.isSelected
         }
-        tab.url = tabData.url
-        tab.order = tabData.order
-        tab.title = tabData.title
-        tab.urlHistorySnapshot = tabData.history as NSArray
-        tab.urlHistoryCurrentIndex = tabData.historyIndex
-        tab.isSelected = tabData.isSelected
-        
-        DataController.save(context: context)
-        
-        return tab
     }
     
     public class func saveScreenshotUUID(_ uuid: UUID?, tabId: String?) {
-        let context = DataController.newBackgroundContext()
-        let tabMO = TabMO.get(fromId: tabId, context: context)
-        tabMO?.screenshotUUID = uuid?.uuidString
-        DataController.save(context: context)
+        DataController.performTask { context in
+            let tabMO = TabMO.get(fromId: tabId, context: context)
+            tabMO?.screenshotUUID = uuid?.uuidString
+        }
+    }
+    
+    public class func saveTabOrder(tabIds: [String]) {
+        DataController.performTask { context in
+            for (i, tabId) in tabIds.enumerated() {
+                guard let managedObject = TabMO.get(fromId: tabId, context: context) else {
+                    log.error("Error: Tab missing managed object")
+                    continue
+                }
+                managedObject.order = Int16(i)
+            }
+        }
     }
 
     public class func getAll() -> [TabMO] {
         let sortDescriptors = [NSSortDescriptor(key: #keyPath(TabMO.order), ascending: true)]
         return all(sortDescriptors: sortDescriptors) ?? []
     }
     
-    public class func get(fromId id: String?, context: NSManagedObjectContext) -> TabMO? {
+    public class func get(fromId id: String?, context: NSManagedObjectContext? = nil) -> TabMO? {
```

Comment:
This concerns me a bit, as the caller may forget to be explicit. At the _very least_, we should set the default context here (`viewContext`) instead of just passing `nil` through. This way if the implementation of `first` changes for some odd reason, it doesn't recursively impact everything that is also using `get` here.

```suggestion
    public class func get(fromId id: String?, context: NSManagedObjectContext = DataController.viewContext) -> TabMO? {
```